Hakawai/Core/HKWControlFlowPluginProtocols.h

```diff
@@ -48,13 +48,12 @@
 -(void) textViewDidProgrammaticallyUpdate:(UITextView *)textView;
 
 /*!
- If available, this method is called when the text view is about to paste text
+ If available, this method is called when the text view is about to programmatically paste text
 
  @param textView Text view for paste
  @param range Range for paste
- @param isProgrammatic  Whether the pasting is programmatic or not
  */
-- (void)textView:(UITextView *)textView willPasteTextInRange:(NSRange)range isProgrammatic:(BOOL)isProgrammatic;
+- (void)textView:(UITextView *)textView willProgrammaticallyPasteTextInRange:(NSRange)range;
```

Comment:
rather than calling it programmatically pasting, i think it makes more sense to call it internal text view pasting

Reply:
I think willPasteProgrammatically is more correct. Because I think that that’s what the delegate handles actually - if something’s programmatically pasted, we clear the mentions formatting at that range
it doesn’t really matter whether it’s an ‘internal’ paste - the plugin doesn’t care where it comes from, it just cares that it’s a paste and it’s not gonna be captured by shouldChangeTextInRange
so I think if there’s a lack of clarity around what “internal pasting” is, it should be clarified via comments/variable names/func names in the textView, but the controlFlowPlugin protocol name should just be about what the control flow plugin cares abt

Reply:
gonna call it custom pasting